lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -238,6 +239,18 @@ fn parse_dogstatsd_metric<'a>(
                     let (_, external_data) = all_consuming(preceded(tag("e:"), external_data)).parse(chunk)?;
                     maybe_external_data = Some(external_data);
                 }
+                // Cardinality: client-provided cardinality for the metric.
+                b'c' if chunk.starts_with(message::CARDINALITY_PREFIX) => {
+                    let (_, cardinality) = all_consuming(preceded(
+                        tag(message::CARDINALITY_PREFIX),
+                        alt((tag("none"), tag("low"), tag("orchestrator"), tag("high"))),
```

Comment:
I would bundle this into a standalone function, similar to `external_data`, so that we don't have to concern `saluki-context` with using `simdutf8` or parsing from `&[u8]`, etc.